qiskit/dagcircuit/dagcircuit.py

```diff
@@ -496,7 +496,7 @@ def _add_op_node(self, op, qargs, cargs):
         self._increment_op(op)
         return node_index
 
-    def _copy_circuit_metadata(self):
+    def copy_circuit_metadata(self):
         """Return a copy of source_dag with metadata but empty."""
```

Comment:
Since this is public now, do you think we should expand this docstring a bit (since it will be published)? Maybe document which attributes are copied or something.

Reply:
actually, now we have a metadata attribute. Maybe we can call the method `copy_empty`?

Reply:
I personally find `empty_like` a bit confusing, since there is no verb in the method name I wouldn't know what it's doing (my first guess would be it's a boolean check to see if the number of elements was near zero). What about `empty_like_copy` or `copy_empty_like`?

Reply:
Ill go for `copy_empty_like`